sim.py

```diff
@@ -217,9 +241,9 @@ def main(argv: List[str]) -> int:
             dispstr = f"CAPTURED + COMPOSITED (DUPS) @ otime {frame.composite_t_ms:0.3f}ms"
         else:
             dispstr = frame.disposition.name
-        print(f"pframe {frame.present_frame} @ {frame.present_t_ms:0.3f}ms, {dispstr}")
+        frame_detail_print(f"pframe {frame.present_frame} @ {frame.present_t_ms:0.3f}ms, {dispstr}")
```

Comment:
In this case, rather than moving the conditional logic to a function, I would just wrap this loop in one big `if not args.stats_only` so that we don't have to perform the loop _at all_, rather than still going through the loop a few thousand times (and making a function call each time) and then throwing away the output at the last moment.

Reply:
Ah, I saw the "Output/Composited Frames" section produced data used for the final stats, but I didn't notice this "Presented Frames" section only makes data for itself to use / is self-contained.